config-object-store/src/main/java/org/hypertrace/config/objectstore/IdentifiedFilterPushedDownObjectStore.java

```diff
@@ -26,17 +27,43 @@ protected IdentifiedFilterPushedDownObjectStore(
         clientConfig);
   }
 
-  public List<ContextualConfigObject<T>> getAllObjects(
-      RequestContext context, F filterInput, Pagination pagination) {
+  public List<ContextualConfigObject<T>> getMatchingObjects(
+      RequestContext context, F filterInput, List<S> sortInput, Pagination pagination) {
     Filter filter = buildFilter(filterInput);
-    List<SortBy> sortByList = buildSortByList(filterInput);
-    return getAllObjectsWithFilter(context, filter, pagination, sortByList);
+    List<SortBy> sortByList = buildSort(sortInput);
+    return getMatchingObjectsWithFilter(context, filter, pagination, sortByList);
   }
 
-  protected abstract Filter buildFilter(F filterInput);
+  public List<T> getMatchingConfigData(
+      RequestContext context, F filterInput, List<S> sortInput, Pagination pagination) {
+    return getMatchingObjects(context, filterInput, sortInput, pagination).stream()
+        .map(ConfigObject::getData)
+        .collect(Collectors.toUnmodifiableList());
+  }
+
+  public Optional<ContextualConfigObject<T>> getObject(
+      RequestContext context, String id, F filter) {
+    return getObject(context, id).flatMap(configObject -> filterObject(configObject, filter));
+  }
 
-  /** Constructs a list of SortBy from client filter or returns an empty list. */
-  protected List<SortBy> buildSortByList(F filterInput) {
-    return Collections.emptyList(); // Override if needed
+  public Optional<T> getData(RequestContext context, String id, F filter) {
+    return getObject(context, id, filter).map(ConfigObject::getData);
   }
+
+  private Optional<ContextualConfigObject<T>> filterObject(
+      ContextualConfigObject<T> configObject, F filter) {
+    return filterConfigData(configObject.getData(), filter)
+        .map(filteredData -> updateConfigData(configObject, filteredData));
+  }
+
+  private ContextualConfigObject<T> updateConfigData(
+      ContextualConfigObject<T> configObject, T updatedData) {
+    return ((ContextualConfigObjectImpl) configObject).toBuilder().data(updatedData).build();
+  }
+
+  protected abstract Optional<T> filterConfigData(T data, F filter);
```

Comment:
We don't need this one or L53, since we've replaced it with the pushed down filter

Reply:
Ok